quil-rs/src/expression/mod.rs

```diff
@@ -125,6 +127,10 @@ impl PartialEq for Expression {
     }
 }
 
+fn are_equal_or_nan(left: f64, right: f64) -> bool {
+    (left.is_nan() && right.is_nan()) || (left == right)
```

Comment:
In principle, I would prefer that `NaN != NaN` (or rather `NaN != x` for any expression `x`. That aligns with the IEEE specification of `NaN` for floats that most languages implement. As a Quil user, I would be surprised to see that `NaN` evaluated as equivalent to anything else.

Is there a practical reason I'm missing as to why we would want this?

Reply:
The only reason I made this change was because proptest generated test input that yielded NaN; the test simplified the expression to NaN, and there was an assertion that the two should be equal.

I guess a better tact here, one that would not impact non-test code, would be to simply filter out proptest input that generates NaN.